node/node.d.ts

```diff
@@ -2189,63 +2201,86 @@ declare module "crypto" {
         update(data: string, input_encoding: "binary" | "base64" | "hex", output_encoding: "utf8" | "ascii" | "binary"): string;
         final(): Buffer;
         final(output_encoding: string): string;
-        setAutoPadding(auto_padding: boolean): void;
+        setAutoPadding(auto_padding?: boolean): void;
         setAuthTag(tag: Buffer): void;
+        setAAD(buffer: Buffer): void;
     }
     export function createSign(algorithm: string): Signer;
     export interface Signer extends NodeJS.WritableStream {
-        update(data: any): void;
-        sign(private_key: string): Buffer;
-        sign(private_key: string, output_format:  'latin1' | 'hex' | 'base64'): string;
+        update(data: string | Buffer): Signer;
+        update(data: string | Buffer, input_encoding: "utf8" | "ascii" | "latin1"): Signer;
+        sign(private_key: string | { key: string; passphrase: string }): Buffer;
+        sign(private_key: string | { key: string; passphrase: string }, output_format: "latin1" | "hex" | "base64"): string;
     }
     export function createVerify(algorith: string): Verify;
     export interface Verify extends NodeJS.WritableStream {
-        update(data: any): void;
+        update(data: string | Buffer): Verify;
+        update(data: string | Buffer, input_encoding: "utf8" | "ascii" | "latin1"): Verify;
         verify(object: string, signature: Buffer): boolean;
-        verify(object: string, signature: string, signature_format: 'latin1' | 'hex' | 'base64'): boolean;
+        verify(object: string, signature: string, signature_format: string): boolean;
```

Comment:
Why the type of `signature_format` was reduced to `string`? the doc still says it is 'latin1', 'hex' or 'base64'